EarlGrey/Synchronization/GREYUIThreadExecutor.m

```diff
@@ -276,6 +257,34 @@ - (BOOL)grey_isTrackingIdlingResource:(id<GREYIdlingResource>)idlingResource {
 #pragma mark - Private
 
 /**
+ *  Register the specified @c resource to be checked for idling before executing test actions.
+ *  A strong reference is held to @c resource until it is deregistered using
+ *  @c grey_deregisterIdlingResource. It is safe to call this from any thread.
+ *
+ *  @param resource The idling resource to register.
+ */
+- (void)grey_registerIdlingResource:(id<GREYIdlingResource>)resource {
```

Comment:
we plan to clean this up and expose it as public API. there are quite a few usages of this internally so please don't rename this.